sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -159,7 +170,10 @@ case class InSubqueryExec(
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     prepareResult()
-    InSet(child, result.toSet).doGenCode(ctx, ev)
+    predicate match {
```

Comment:
Can we do `predicate.doGenCode(ctx, ev)`?

Reply:
I tried `predicate.doGenCode()` first, but `doGenCode()` in `predicate` is protected. So we can not access it here.

Reply:
Could you leave some cmments here about why we cannot do `predicate.doGenCode(ctx, ev)`?